Documentation/coding-guidelines/project-guidelines.md

```diff
@@ -1,50 +1,84 @@
+#Build Project Guidelines
+In order to work in corefx repo you must first run build.cmd/sh from the root of the repo at least
+once before you can iterate and work on a given library project.
+
+##Behind the scenes with build.cmd
+
+- Setup tools (currently done in init-tools but will later be a boot-strap script in run.cmd/sh)
+- Restore external dependencies
+ - CoreCLR - Copy to bin\runtime\[BuildTarget]
+ - Netstandard Library - Copy to bin\ref\netstandard
+ - UAP - Copy to bin\runtime\uap??
+- Build targeting pack
+ - Build src\ref.builds file which will build all the references assemblies to bin\ref\[BuildTargetFramework]. Where BuildTargetFramework is a versionless TFM representing the
+ vertical build (e.g. netcoreapp/netfx/uap)
+- Build product
+ - Build src\src.builds file which will build all the source projects
+ - Source projects will have <Reference> to their dependencies or in some
+   case P2P references to other source projects when necessary (i.e. partial facades)
+ - Source projects will copy output to bin\runtime\[BuildConfiguration]
+- Tests will be built as part of build-tests.cmd and will require that build.cmd has
+  been ran at least once.
+- Packages will be built as part of build-packages.cmd and will require that build.cmd has
+  been run at least once.
+
+##Behind the scenes with build-test.cmd
+TODO:
+
+##Behind the scenes with build-packages.cmd
+TODO:
+
+#Build Pivots
+Below is a list of all the various options we pivot the project builds on.
 
-In the ref directory for the library there should be at most **one** `.csproj` that contains the latest API for the reference assembly for the library.  If a library cannot support all supported targets using the latest API it should use `.depproj` projects for each significant historical version in order to download and redistribute the historical reference assembly in the latest package.  If a library is a pure portable library with a single implementation it need not use a reference assembly at all. (see [Reference assembly projects](#reference-assembly-projects)).
+- **Flavor:** Debug, Release
+- **Architecture:** x86, x64, ARM, ARM64, AnyCPU
+- **OS:** Windows_NT, Linux, OSX, FreeBSD, AnyOS
+- **Target Frameworks:** NetFx (aka Desktop), netstandard (aka dotnet/Portable), NETCoreApp (aka .NET Core), UAP (aka UWP/Store/netcore50), [??Should we have any target framework??]
+- **Platform Runtimes:** [??Should we remove this option??] NetFx (aka CLR/Desktop), CoreCLR, CoreRT (aka NetNative/AOT/MRT)
+
+##Individual build properties
+The following are the properties associated with each build pivot
```

Comment:
I don't think TestTFM will be relevant any longer. You will only be able to run on the runtime folder that you built from the vertical so if your BuildConfiguration is netcoreapp* then it will run on that platform but if it is netfx or uap it will run on that platform. So it will be controlled by the BuildConfiguration (I will get to that in the document when I fill in the test part).